vizro-core/src/vizro/models/_components/ag_grid.py

```diff
@@ -0,0 +1,121 @@
+import logging
+from typing import Dict, List, Literal
+
+import dash_ag_grid as dag
+import pandas as pd
+from dash import State, dcc, html
+
+try:
+    from pydantic.v1 import Field, PrivateAttr, validator
+except ImportError:  # pragma: no cov
+    from pydantic import Field, PrivateAttr, validator
+
+import vizro.tables as vt
+from vizro.actions._actions_utils import CallbackTriggerDict, _get_component_actions, _get_parent_vizro_model
+from vizro.managers import data_manager
+from vizro.models import Action, VizroBaseModel
+from vizro.models._action._actions_chain import _action_validator_factory
+from vizro.models._components._components_utils import _process_callable_data_frame
+from vizro.models._models_utils import _log_call
+from vizro.models.types import CapturedCallable
+
+logger = logging.getLogger(__name__)
+
+
+class AgGrid(VizroBaseModel):
+    """Wrapper for `dash-ag-grid.AgGrid` to visualize grids in dashboard.
+
+    Args:
+        type (Literal["ag_grid"]): Defaults to `"ag_grid"`.
+        figure (CapturedCallable): AgGrid like object to be displayed. For more information see:
+            [`dash-ag-grid.AgGrid`](https://dash.plotly.com/dash-ag-grid).
+        title (str): Title of the table. Defaults to `""`.
+        actions (List[Action]): See [`Action`][vizro.models.Action]. Defaults to `[]`.
+
+    """
+
+    type: Literal["ag_grid"] = "ag_grid"
+    figure: CapturedCallable = Field(..., import_path=vt, description="AgGrid to be visualized on dashboard")
+    title: str = Field("", description="Title of the AgGrid")
+    actions: List[Action] = []
+
+    _callable_object_id: str = PrivateAttr()
+
+    # Component properties for actions and interactions
+    _output_property: str = PrivateAttr("children")
+
+    # validator
+    set_actions = _action_validator_factory("cellClicked")
+    _validate_callable = validator("figure", allow_reuse=True, always=True)(_process_callable_data_frame)
+
+    # Convenience wrapper/syntactic sugar.
+    def __call__(self, **kwargs):
+        kwargs.setdefault("data_frame", data_manager._get_component_data(self.id))
+        return self.figure(**kwargs)
+
+    # Convenience wrapper/syntactic sugar.
+    def __getitem__(self, arg_name: str):
+        # See table implementation for more details.
+        if arg_name == "type":
+            return self.type
+        return self.figure[arg_name]
+
+    # Interaction methods
+    @property
+    def _filter_interaction_input(self):
+        """Required properties when using pre-defined `filter_interaction`."""
+        return {
+            "cellClicked": State(component_id=self._callable_object_id, component_property="cellClicked"),
+            "modelID": State(component_id=self.id, component_property="id"),  # required, to determine triggered model
+        }
```

Comment:
Do we prefer comments above the line which it points to?
Same for the `vm.Table`.

Reply:
Not sure - do you have a preference? If it fits inline I thought why not, but not sure if there is best practice on that